concurrent/src/test/java/io/airlift/concurrent/TestAsyncSemaphore.java

```diff
@@ -289,122 +288,135 @@ public void testNoStackOverflow()
         Futures.allAsList(futures).get(1, TimeUnit.MINUTES);
     }
 
-    @DataProvider
-    public static Object[][] concurrency()
+    public static int[] concurrency()
     {
-        return new Object[][] {{1}, {2}, {3}, {5}};
+        return new int[] {1, 2, 3, 5};
     }
 
-    @Test(dataProvider = "concurrency")
-    public void testProcessAllEmptyList(int concurrency)
+    @Test
+    public void testProcessAllEmptyList()
             throws Exception
     {
-        ListenableFuture<List<Object>> result = AsyncSemaphore.processAll(ImmutableList.of(), (i) -> immediateCancelledFuture(), concurrency, directExecutor());
-        assertThat(result).isDone();
-        assertThat(result.get()).isEqualTo(ImmutableList.of());
+        for (int concurrency : concurrency()) {
+            ListenableFuture<List<Object>> result = AsyncSemaphore.processAll(ImmutableList.of(), (i) -> immediateCancelledFuture(), concurrency, directExecutor());
+            assertThat(result).isDone();
+            assertThat(result.get()).isEqualTo(ImmutableList.of());
+        }
     }
 
-    @Test(dataProvider = "concurrency")
-    public void testProcessAllSingleCallable(int concurrency)
+    @Test
+    public void testProcessAllSingleCallable()
             throws Exception
     {
-        SettableFuture<String> future = SettableFuture.create();
-        ListenableFuture<List<String>> result = AsyncSemaphore.processAll(ImmutableList.of(1), (i) -> future, concurrency, directExecutor());
-        assertThat(result).isNotDone();
-        future.set("value");
-        assertThat(result).isDone();
-        assertThat(result.get()).isEqualTo(ImmutableList.of("value"));
+        for (int concurrency : concurrency()) {
+            SettableFuture<String> future = SettableFuture.create();
+            ListenableFuture<List<String>> result = AsyncSemaphore.processAll(ImmutableList.of(1), (i) -> future, concurrency, directExecutor());
+            assertThat(result).isNotDone();
+            future.set("value");
+            assertThat(result).isDone();
+            assertThat(result.get()).isEqualTo(ImmutableList.of("value"));
```

Comment:
assertions should include value of `concurrency` 
(but i'd recommend using junit builtin support for parameters provider instead)